lib/flow/VaporField.cpp

```diff
@@ -237,10 +237,15 @@ int VaporField::GetVelocity(double time, const glm::vec3 &pos, glm::vec3 &veloci
             missingV[i] = grid->GetMissingValue();
         }
         auto  hasMissing = glm::equal(velocity, missingV);
+        // If missing values are represented using NaN, you cannot compare equality with them!
+        for (int i = 0; i < 3; i++) {
+          if (std::isnan(missingV[i]) && velocity[i])
```

Comment:
I would try and avoid implicitly converting float to bool.

Reply:
Good catch! It was a typo and is fixed in be4afab5c .